velox/functions/sparksql/GetJsonObject.h

```diff
@@ -148,35 +150,28 @@ struct GetJsonObjectFunction {
       // can check the validity of ending character.
       case simdjson::ondemand::json_type::number: {
         switch (rawResult.get_number_type()) {
-          case simdjson::ondemand::number_type::unsigned_integer: {
-            uint64_t numberResult;
-            if (!rawResult.get_uint64().get(numberResult)) {
-              ss << numberResult;
-              result.append(ss.str());
-              return true;
-            }
-            return false;
-          }
-          case simdjson::ondemand::number_type::signed_integer: {
-            int64_t numberResult;
-            if (!rawResult.get_int64().get(numberResult)) {
-              ss << numberResult;
-              result.append(ss.str());
-              return true;
-            }
-            return false;
-          }
           case simdjson::ondemand::number_type::floating_point_number: {
             double numberResult;
             if (!rawResult.get_double().get(numberResult)) {
-              ss << rawResult;
-              result.append(ss.str());
+              result.append(
+                  util::Converter<TypeKind::VARCHAR>::tryCast(numberResult)
+                      .value());
               return true;
             }
             return false;
           }
-          default:
-            VELOX_UNREACHABLE();
+          default: {
+            std::string_view intResult = trimToken(rawResult.raw_json_token());
+            // Spark uses Jackson to parse JSON, which does not preserve the
+            // negative sign for -0. See the implementation here:
+            // https://github.com/FasterXML/jackson-core/blob/93deb382b962359dff28a312741eff62c2573e75/src/main/java/com/fasterxml/jackson/core/util/TextBuffer.java#L699-L702
```

Comment:
Which version does the commit hash ID point to? It would be better to use the link below that points to 2.19.0 used by latest Spark.

https://github.com/FasterXML/jackson-core/blob/jackson-core-2.19.0/src/main/java/com/fasterxml/jackson/core/util/TextBuffer.java#L699-L702

Reply:
Addressed, change to 2.19.2 that's used in latest Spark.